sha2/src/consts.rs

```diff
@@ -163,6 +163,56 @@ pub const K64X2: [[u64; 2]; 40] = [
     [K64[79], K64[78]],
 ];
 
+macro_rules! dup_array {
+    ([$([$a:expr, $b:expr]),*,]) => {[
+        $($b, $a, $b, $a),*,
+    ]}
+}
+
+/// Constants necessary for SHA-512 family of digests.
+pub const K64X4: [u64; 160] = dup_array!([
+    [K64[1], K64[0]],
```

Comment:
I wonder if we can remove this duplication without negatively impacting performance. i think we can do necessary shuffling in one-two instructions. Yes, it's more than zero, but it would mean less memory pressure.

Also if we are to keep this table, then we could enforce it's alignment to 32 bytes, so we will be able to use `_mm256_load_si256` (aligned load) instead of `_mm256_loadu_si256` (unaligned load). Ideally this table would contain `__m256i`, but unfortunately `_mm256_set_epi64x` is not `const`ified, so we would have to use `transmute_copy` for it.

Reply:
Checked with this wrapper for alignment:
```rust
#[repr(align(32))]
pub struct Align32;

pub struct AlignedArray<A, T> {
    _alignment: [A; 0],
    value: T,
}

impl<A, T> core::ops::Deref for AlignedArray<A, T> {
    type Target = T;

    #[inline]
    fn deref(&self) -> &T {
        &self.value
    }
}

pub const K64X4: AlignedArray<Align32, [u64; 160]> = AlignedArray { .... };
```

The difference is insignificant so removed this array completely